crates/bevy_reflect/src/lib.rs

```diff
@@ -651,6 +653,10 @@ pub mod __macro_exports {
         DynamicTupleStruct, GetTypeRegistration, TypeRegistry,
     };
 
+    pub use alloc::borrow::Cow;
+    pub use alloc::boxed::Box;
+    pub use alloc::string::ToString;
```

Comment:
Agreed, I've moved these items into a sub-module `alloc_utils` (instead of `alloc` since the structure of the crate `alloc` is not the same as `alloc_utils`) and included a short comment explaining that the `alloc` namespace is required for `no_std` support, but isn't always available.